hw/ip_templates/rv_core_ibex/doc/boot-rom-patching.md

```diff
@@ -0,0 +1,476 @@
+It provides the framework necessary to bootstrap the SoC computing elements in a secure and trusted fashion.
+This document covers the changes necessary for the OpenTitan to boot first while also allowing an SoC with integrated OpenTitan to add small customizations to the boot path configuration that fit with the SoC topology.
+A few examples are voltage ramp up sequence, clock settings etc.
+Such customizations are expected to
+
+- Be non-proprietary such that they can be made available in the open source environment for audit.
+- Only require minimal flow to enable a path to external flash that is under OpenTitan control.
+
+Flows involving reset and bring up for the rest of the SoC, which may be potentially more complex and involve proprietary code, are expected to be handled by other computing element(s) that are bootstrapped by the integrated OpenTitan and running authenticated / measured code.
+
+This proposal introduces a second OpenTitan boot ROM partition with efuse/OTP based patching mechanism that enables each SoC to add such small customizations and provide critical survivability hooks while also removing the dependency on the integrated flash component.
+Integrated flash technology may not be available for SoCs built using advanced process nodes.
+
+Two models for OpenTitan based ROM patch deployment are envisioned:
+
+- **Factory patching** - here the ROM patches are programmed into the eFuses in the factory.
+  This may be to accommodate post-silicon bug fixes without needing a new step of the silicon.
+  Performed through the secure manufacturing flow.
+- **In-field patching** - The capability may be used to recover parts in the field but that requires an ability to burn eFuses in the field.
+
+Note that patch structure for both deployment models is the same.
+An infield patch can be viewed as a new revision of the OTP patch.
+The patch in eFuses will be signed using an appropriate PK signing algorithm
+
+### Programming method:
+
+- If prior to the PROD lifecycle stage: Using the manufacturing efuse programming interface.
+- In PROD lifecycle stage: Secure, verified patch update programming managed by secure application running on OpenTitan Ibex core (e.g., update patch revision in factory after provisioning, backup in-field patching mechanism).
+
+The factory provisioning model is expected to be the mainstream deployment model that covers bugs found throughout the product development lifecycle until product deployment phase.
+The in-field patching capability is primarily a backup mechanism used only in very rare cases to patch a critical ROM bug escape in the field, especially some hard to catch security or functional bugs.
+**It is advised to avoid reliance on the in-field ROM patch as much as possible.**
+
+This document also contains a detailed specification for the OpenTitan ROM patching architecture, the programming interfaces, and example sequences for patching ROM firmware.
```

Comment:
The "also" here is a bit odd, because it looks like it applies to the advice on the previous line, which isn't what you mean. I think there's also a slight problem with the English (because it looks like we're claiming a detailed specification of example sequences). I'd suggest being a bit more explicit at the start and turning the list into bullets:

```
As well as describing the form of a ROM patch, this document contains:
- A specification of the OpenTitan ROM patching architecture.
- A description of how to use a ROM patch.
- Example sequences for patching ROM firmware.
```

(I'm not completely sure about item 2, though! This was me trying to guess what "the programming interfaces" meant. Since this ends up being a mini-TOC anyway, maybe a couple of internal links would help).